node/src/components/binary_port.rs

```diff
@@ -794,6 +797,72 @@ where
             };
             BinaryResponse::from_value(status, protocol_version)
         }
+        InformationRequest::Reward {
+            era_identifier,
+            validator,
+            delegator,
+        } => {
+            let Some(header) = resolve_era_switch_block_header(effect_builder, era_identifier).await else {
+                return BinaryResponse::new_error(ErrorCode::SwitchBlockNotFound, protocol_version);
+            };
+            let Some(parent_header) =
+                effect_builder.get_block_header_from_storage(*header.parent_hash(), true).await else {
+                return BinaryResponse::new_empty(protocol_version);
+            };
```

Comment:
If there is no parent, you're at genesis and if there isn't an entry for era 0 in the snapshot (there might be, but I'm not sure), you can just use the one for era 1 - they will always be identical, anyway.

Reply:
Although, I guess another valid reason is that the switch block is the earliest block we have in storage? In which case we won't be able to serve the request. And if the switch block is genesis, there are also no rewards to be returned. So maybe it's actually okay to return an empty response in this case.

Reply:
Yeah, I returned empty response here because there wouldn't be any rewards in genesis
In case of a switch block that has no parent in storage, it could maybe be worth adjusting this to return an error